### PR TITLE
Add telemetry filters

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java [hunk 1]
```diff
@@ -308,24 +379,24 @@ public String toString() {
...
+        public MatchType matchType;
         public List<String> spanNames = new ArrayList<>();
         public List<String> logNames = new ArrayList<>();
         public List<ProcessorAttribute> attributes = new ArrayList<>();
 
         public void validate(ProcessorType processorType, IncludeExclude includeExclude) throws FriendlyException {
             if (matchType == null) {
                 throw new FriendlyException(processorType.anX + " processor configuration has an " + includeExclude + " section that is missing a \"matchType\".",
-                        "Please provide \"matchType\" under the include/exclude section of the " + processorType + " processor configuration. " +
+                        "Please provide a \"matchType\" under the " + includeExclude + " section of the " + processorType + " processor configuration. " +
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java:390]: includeExclude, typo?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    private final List<MatcherGroup> matcherGroups;
+
+    SamplingOverrides(List<SamplingOverride> overrides) {
+        matcherGroups = new ArrayList<>();
+        for (SamplingOverride override : overrides) {
+            matcherGroups.add(new MatcherGroup(override));
+        }
+    }
+
+    // returns -1 if no override
```
[the remark this answers is not in the document]
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java]: 😂 no it does not, removed

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import com.microsoft.applicationinsights.agent.Exporter;
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration.MatchType;
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration.SamplingOverride;
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration.SamplingOverrideAttribute;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+class SamplingOverrides {
```
[the remark this answers is not in the document]
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java:18]: ya, I agree this is not great, typically in Java naming pattern `SamplingOverrides` would be a static util class for working with `SamplingOverride`. I added a TODO on this class to come up with something better.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java:390]: this is a variable `includeExclude` whose `toString()` will return one of "include" or "exclude"

**Review by heyams — APPROVED** [review]
👍

**Pull request merged**